capstone-frontend/src/components/modal.jsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState, useContext, useRef, createContext } from "react";
+import ApplicationContext from "../applicationContext";
+import Modal from "react-modal";
+
+const customStyles = {
+    content: {
+      top: "50%",
+      left: "50%",
+      marginRight: "-50%",
+      transform: "translate(-50%, -50%)",
+    },
+  };
+
+function ModalPopulate() {
+    const { modalIsOpen, setIsOpen, socket } = useContext(ApplicationContext);
+    const websocket = socket.current;
+    
+    function openModal() {
+      setIsOpen(true);
+    }
+    function closeModal() {
+      setIsOpen(false);
+    }
+    let userId = null;
```

Comment:
I think this should be react state and not a local variable, since you want to change what you're rendering when this value changes (i.e. change the contents of the modal)